source/macos/AdaptiveCards/AdaptiveCardsTests/Renderers/InputDateRendererTests.swift

```diff
@@ -31,20 +31,55 @@ class InputDateRendererTest: XCTestCase {
     }
 
     func testRendererForMinValue() {
-        let minVal: String = "23/11/1999"
+        let minVal: String = "1999-03-23"
         inputDate = .make(min: minVal)
 
         let inputDateField = renderDateInput()
         XCTAssertEqual(inputDateField.minDateValue, minVal)
     }
 
     func testRendererForMaxValue() {
-        let maxValue: String = "23/11/2022"
+        let maxValue: String = "2022-02-20"
         inputDate = .make(max: maxValue)
 
         let inputDateField = renderDateInput()
         XCTAssertEqual(inputDateField.maxDateValue, maxValue)
     }
+    
+    func testClearsText() {
+        let val: String = "2000-11-23"
+        inputDate = .make(value: val)
+
+        let inputDateField = renderDateInput()
+        inputDateField.clearButton.performClick()
+        XCTAssertEqual(inputDateField.dateValue, "")
+        XCTAssertTrue(inputDateField.clearButton.isHidden)
+    }
+    
+    func testClearButtonHiddenWithPlaceholder() {
+        let placeholderString: String = "Sample Placeholder"
+        inputDate = .make(placeholder: placeholderString)
+
+        let inputDateField = renderDateInput()
+        inputDateField.clearButton.performClick()
+        XCTAssertTrue(inputDateField.clearButton.isHidden)
+    }
+    
+    func testClearButtonHiddenWithNoText() {
+        inputDate = .make()
+        
+        let inputDateField = renderDateInput()
+        inputDateField.clearButton.performClick()
+        XCTAssertTrue(inputDateField.clearButton.isHidden)
```

Comment:
Same here. Why is `performClick()` needed??
It shouldn't be called as button will be already hidden. Just checking whether it's hidden is enough.

Reply:
True
Will change